HVSample/src/main/java/com/microsoft/hsg/android/hvsample/FileUploadActivity.java

```diff
@@ -31,191 +31,199 @@
+		mClient = new HealthVaultClient();
+
+		mProgressContainer = findViewById(R.id.camera_progress_container);
+		mProgressContainer.setVisibility(View.INVISIBLE);
+
+		mSurfaceView = (SurfaceView)findViewById(R.id.camera_surfaceView);
+		SurfaceHolder holder = mSurfaceView.getHolder();
+
+		holder.addCallback(new SurfaceHolder.Callback() {
 
-            public void surfaceCreated(SurfaceHolder holder) {
-                try {
-                    if (camera != null) {
-                        camera.setPreviewDisplay(holder);
-                    }
-                } catch (IOException exception) {
-                    exception.printStackTrace();
-                }
-            }
+			public void surfaceCreated(SurfaceHolder holder) {
+				try {
+					if (mCamera != null) {
+						mCamera.setPreviewDisplay(holder);
+					}
+				} catch (IOException exception) {
+					exception.printStackTrace();
+				}
+			}
 
-            public void surfaceDestroyed(SurfaceHolder holder) {
-                if (camera != null) {
-                    camera.stopPreview();
-                }
-            }
+			public void surfaceDestroyed(SurfaceHolder holder) {
+				if (mCamera != null) {
+					mCamera.stopPreview();
+				}
+			}
 
-            public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
-            	if (camera == null) return;
-            	
-                Camera.Parameters parameters = camera.getParameters();
-                Size s = getBestSupportedSize(parameters.getSupportedPreviewSizes(), w, h);
-                parameters.setPreviewSize(s.width, s.height);
-                s = getBestSupportedSize(parameters.getSupportedPictureSizes(), w, h);
-                parameters.setPictureSize(s.width, s.height);
-                camera.setParameters(parameters);
-                try {
-                    camera.startPreview();
-                } catch (Exception e) {
-                    camera.release();
-                    camera = null;
-                }
-            }
-        });
+			public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
+				if (mCamera == null) return;
+				Camera.open();
+				Camera.Parameters parameters = mCamera.getParameters();
+				Size s = getBestSupportedSize(parameters.getSupportedPreviewSizes(), w, h);
+				parameters.setPreviewSize(s.width, s.height);
+				s = getBestSupportedSize(parameters.getSupportedPictureSizes(), w, h);
+				parameters.setPictureSize(s.width, s.height);
+				mCamera.setParameters(parameters);
+				try {
+					mCamera.startPreview();
+				} catch (Exception e) {
+					mCamera.release();
+					mCamera = null;
+				}
+			}
+		});
 	
 		// write buttons
 		wireButtons();
 	}
-	
-	
-	
+
 	@Override
 	protected void onStart() {
-		// TODO Auto-generated method stub
 		super.onStart();
-		hvClient.start();
+		mClient.start();
 	}
 
-
-
 	@Override
 	protected void onStop() {
-		// TODO Auto-generated method stub
 		super.onStop();
```

Comment:
In a previous onStop you did client.stop() first and then super.onStop() but here you are calling super first. Don't you want to be consistent? super.onStart() client.start() cliet.stop() super.stop()?